tests/unit/browser/test_mhtml.py

```diff
@@ -0,0 +1,289 @@
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20h
+        -----=_qute-UUID
+        Content-Location: http://i.example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20i
+        -----=_qute-UUID
+        Content-Location: http://t.example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20t
+        -----=_qute-UUID
+        Content-Location: http://z.example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20z
+        -----=_qute-UUID--
+        """)
+
+
+def test_empty_content_type(checker):
+    writer = mhtml.MHTMLWriter(root_content=b'',
+                               content_location='http://example.com/',
+                               content_type='text/plain')
+    writer.add_file('http://example.com/file', b'file content')
+    writer.write_to(checker.fp)
+    checker.expect("""
+        Content-Type: multipart/related; boundary="---=_qute-UUID"
+        MIME-Version: 1.0
+
+        -----=_qute-UUID
+        Content-Location: http://example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+
+        -----=_qute-UUID
+        MIME-Version: 1.0
+        Content-Location: http://example.com/file
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20content
+        -----=_qute-UUID--
+        """)
+
+
+def test_removing_file_from_mhtml(checker):
+    writer = mhtml.MHTMLWriter(root_content=b'root',
+                               content_location='http://example.com/',
+                               content_type='text/plain')
+    writer.add_file('http://evil.com/', b'file content')
+    writer.remove_file('http://evil.com/')
+    writer.write_to(checker.fp)
+    checker.expect("""
+        Content-Type: multipart/related; boundary="---=_qute-UUID"
+        MIME-Version: 1.0
+
+        -----=_qute-UUID
+        Content-Location: http://example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        root
+        -----=_qute-UUID--
+        """)
+
+
+@pytest.mark.parametrize('style, expected_urls', [
+    ("@import 'default.css'", ['default.css']),
+    ('@import "default.css"', ['default.css']),
+    ("@import \t 'tabbed.css'", ['tabbed.css']),
+    ("@import url('default.css')", ['default.css']),
+    ("""body {
+    background: url("/bg-img.png")
+    }""", ['/bg-img.png']),
+    ('background: url(folder/file.png)', ['folder/file.png']),
+    ('content: url()', []),
+])
+def test_css_url_scanner(style, expected_urls):
+    expected_urls.sort()
+    urls = mhtml._get_css_imports(style)
+    urls.sort()
+    assert urls == expected_urls
+
+
+def test_noclose_bytesio_fake_close():
```

Comment:
For the pylint failures, see [pylint issue 540](https://bitbucket.org/logilab/pylint/issues/540/no-member-for-ioiobase-subclass) - after moving it to a class, please add something like this to the class body:

```python
# WORKAROUND for https://bitbucket.org/logilab/pylint/issues/540/
# pylint: disable=no-member
```